controllers/recipesController.js

```diff
@@ -0,0 +1,59 @@
+import ctrlWrapper from "../decorators/ctrlWrappe.js";
+import HttpError from "../helpers/HttpError.js";
+
+import {
+  getRecipes,
+  getRecipe,
+  updateFavorites,
+} from "../services/recipesService.js";
+
+const getPopular = async (req, res) => {
+  const { page = 2, limit = 5 } = req.query;
+  const skip = (page - 1) * limit;
+  const popular = await getRecipes(skip, limit);
+  res.json(popular);
+};
+
+const getFavorite = async (req, res) => {
+  const { _id: userId } = req.user;
+  const { page = 1, limit = 5 } = req.query;
+  const skip = (page - 1) * limit;
+  const favorites = await getRecipes(skip, limit, userId);
+  res.json(favorites);
+};
+
+const addToFavorite = async (req, res) => {
+  const { _id: userId } = req.user;
+  const { id: _id } = req.params;
```

Comment:
First for user, second for recipe. Search recipe by id, then add user id to favorite field inside the recipe object

Reply:
So it would be great to have specific different names for those variables(userId is ok, so we can call the sacond one as recipeId). I will close the issue now, but please keep it in mind for future